src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1032,34 +1083,51 @@ private void RemoveWorseMembers<TMember>(ArrayBuilder<MemberResolutionResult<TMe
             // then we need to do no more work; we know it cannot win. But it is also possible that
             // it is not worse than anything but not better than everything. 
 
+            if (results.Count <= 1)
```

Comment:
Would it make sense to pass around `OverloadResolutionResult<T>` instead so we can define something like `SingleValid()` on it, just like operators?